### PR TITLE
cat: fix #5186 by adding explicit flush.

### LINKED ISSUE #5186: cat: wrong concatenation binary content
```

$ cat -A /proc/1/cmdline /proc/version
/sbin/init^@splash^@Linux version 6.2.0-26-generic (buildd@bos03-amd64-042) (x86_64-linux-gnu-gcc-11 (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2$

$ coreutils cat -A /proc/1/cmdline /proc/version
/sbin/init^@splash^@Linux version 6.2.0-26-generic (buildd@bos03-amd64-042) (x86_64-linux-gnu-gcc-11 (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2$


```

But

```
$ coreutils cat /proc/1/cmdline /proc/version    
Linux version 6.2.0-26-generic (buildd@bos03-amd64-042) (x86_64-linux-gnu-gcc-11 (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2
/sbin/initsplash

$ cat /proc/1/cmdline /proc/version
/sbin/initsplashLinux version 6.2.0-26-generic (buildd@bos03-amd64-042) (x86_64-linux-gnu-gcc-11 (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2

```

### PR DESCRIPTION
**Pull request by gim913**
https://github.com/uutils/coreutils/issues/5186
See linked issue for details

### CODE REVIEW

File: src/uu/cat/src/cat.rs [hunk 1]
```diff
@@ -444,6 +444,7 @@ fn write_fast<R: FdReadable>(handle: &mut InputHandle<R>) -> CatResult<()> {
         }
         stdout_lock.write_all(&buf[..n])?;
     }
+    stdout_lock.flush()?;
```
Comment (sylvestre) [inline src/uu/cat/src/cat.rs:447]: could you please add a comment explaining why the flush is needed?
thanks

File: tests/by-util/test_cat.rs [hunk 2]
```diff
@@ -460,6 +460,20 @@ fn test_dev_full_show_all() {
...
+/// without additional flush output gets reversed.
+#[test]
+#[cfg(any(target_os = "linux"))]
+fn test_write_fast_fallthrough_uses_flush() {
+    const PROC_INIT_CMDLINE: &str = "/proc/1/cmdline";
+
+    new_ucmd!()
+        .args(&[PROC_INIT_CMDLINE, "alpha.txt"])
+        .succeeds()
+        .stdout_only("/sbin/init\0abcde\nfghij\nklmno\npqrst\nuvwxyz\n");
```
Comment (sylvestre) [inline tests/by-util/test_cat.rs:474]: please add these words to the ignore list

**Comment by sylvestre** [comment]
@gim913 ping ?

**Comment by gim913** [comment]
@sylvestre hi, sorry, missed previous message, will try to address this this or next week, thanks 👋

**Comment by sylvestre** [comment]
@gim913 ping ? :)

**Comment by gim913** [comment]
Hi, sorry, life happened, and then rust update forced me to fight with llvm installation 😢
Can't test that spellcheck change, based it on other uses inside the file, hope it's ok.

P.S. want me to rebase and/or squash the branch?

**Comment by gim913** [comment]
@sylvestre ping? ;)

> P.S. want me to rebase and/or squash the branch?

**Review by BenWiederhake — COMMENTED** [review]
The diff looks good to me, and could be merged.

Please rebase on top of current `main`; there are many CI failures, and since this branch is roundabout 150 commits behind it, I'd like to see a (more) successful run first.

**Comment by gim913** [comment]
Hello, are the two failures visible above something known?
I've tried looking at produced android artifacts (screenshots), but they're not super helpful.

**Comment by cakebaker** [comment]
Yes, the Android tests currently also fail for other PRs and hence are unrelated to this PR. Some of the other failing tests, however, are related to your PR :|

**Review by BenWiederhake — APPROVED** [review]
LGTM, remaining CI failures are a known issue (fixed by #6762).

**Pull request merged**